modules/es/fsw/src/cfe_es_cds.c

```diff
@@ -659,7 +659,7 @@ void CFE_ES_FormCDSName(char *FullCDSName, const char *CDSName, CFE_ES_AppId_t T
     AppName[OS_MAX_API_NAME - 1] = '\0';
 
     /* Complete formation of processor specific table name */
-    sprintf(FullCDSName, "%s.%s", AppName, CDSName);
+    snprintf(FullCDSName, CFE_MISSION_ES_CDS_MAX_FULL_NAME_LEN, "%s.%s", AppName, CDSName);
```

Comment:
I don't like this one.  In this case the API/function prototype did not provide a length at all, but assuming the user always has provided a buffer of `CFE_MISSION_ES_CDS_MAX_FULL_NAME_LEN` arguably is just as bad as calling `sprintf()` here.  Note that the full name length is computed to be the worst case length of `CFE_MISSION_ES_CDS_MAX_NAME_LENGTH` and `OS_MAX_API_NAME` put together, so if the buffer actually is at least this length, there is no chance of overflow.

The fundamental flaw of `sprintf()` is that it just assumes the buffer is long enough to hold the result.  Switching this to calling `snprintf()` with an "assumed long enough" length parameter doesn't fix the problem, it is still assuming the length is long enough when it might not actually be.

I think this particular function (CFE_ES_FormCDSName) is a broken API, and should be removed (its not public).  It is called in just one place - from CFE_ES_RegisterCDS() - and that can just do an `snprintf` directly.  It actually needs to know the app name anyway for an event later on, so this would also avoid getting it twice.  A win-win.